src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -211,4 +226,55 @@ private void validateMedicalInventory(MedicalInventory medInventory) throws OHDa
 			throw new OHDataValidationException(errors);
 		}
 	}
+	
+	@Transactional(rollbackFor = OHServiceException.class)
+	public void validateInventory(MedicalInventory inventory, List<MedicalInventoryRow> inventoryRowSearchList) throws OHDataValidationException, OHServiceException {
+		LocalDateTime movFrom = inventory.getLastModifiedDate();
+		LocalDateTime movTo = TimeTools.getNow();
+		List<Movement> movements = movBrowserManager.getMovements(null, null, null, null, movFrom, movTo, null, null, null, null); 
+		boolean updated = false;
+		if (!movements.isEmpty()) {
+			Map<Medical, List<Movement>> groupedByMedical = movements.stream().collect(Collectors.groupingBy(Movement::getMedical));
+			for (Iterator<MedicalInventoryRow> iterator = inventoryRowSearchList.iterator(); iterator.hasNext();) {
+				MedicalInventoryRow medicalInventoryRow = (MedicalInventoryRow) iterator.next();
+				String lotCode = medicalInventoryRow.getLot().getCode();
+				double theoQty = medicalInventoryRow.getTheoreticQty();
+				Medical medical = medicalInventoryRow.getMedical();
+				List<Movement> movs = groupedByMedical.get(medical);
+				if (movs != null) {
+					Movement mov = movs.get(0);
+					String lotCodeOfMovement = mov.getLot().getCode();
+					Lot lot = null;
+					Integer medicalCode = medical.getCode();
+					List<Lot> lots = movStockInsertingManager.getLotByMedical(medical).stream().filter(l -> l.getCode().equals(lotCodeOfMovement)).collect(Collectors.toList());
+					if (lots != null) {
+						lot = lots.get(0);
+						double mainStoreQty = (double)lot.getMainStoreQuantity();
+						Integer movMedicalCode = mov.getMedical().getCode();
+						if (movMedicalCode.equals(medicalCode) && lotCodeOfMovement.equals(lotCode)) {
+							if (mainStoreQty != theoQty) { 
+								medicalInventoryRow.setTheoreticQty(mainStoreQty);
+								medicalInventoryRowManager.updateMedicalInventoryRow(medicalInventoryRow);
+								updated = true;
+							}
+						} else {
+							MedicalInventoryRow invRow = medicalInventoryRowManager.getMedicalInventoryRowByMedicalCodeAndLotCode(medicalCode, lotCode);
+							if (invRow != null) {
+								invRow.setTheoreticQty(mainStoreQty);
+								medicalInventoryRowManager.updateMedicalInventoryRow(invRow);
+							} else {
+								MedicalInventoryRow medInvRow = new MedicalInventoryRow(0, mainStoreQty, mainStoreQty, inventory, medical, lot);
+								medicalInventoryRowManager.newMedicalInventoryRow(medInvRow);
+							}
+						}
+					}
+				}
+			}
+		}
+		if (updated) {
+			List<OHExceptionMessage> errors = new ArrayList<>();
+			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.inventory.sometheoriticqtyhavebeenupdatepleasecontrole.msg")));
+			throw new OHDataValidationException(errors);
+		}
```

Comment:
Instead to have a single generic message, I would StringBuild a message including the generic bundle and appending the list of products/lots that have been actually changed.